XPlat/ExposureNotification/source/Xamarin.ExposureNotification/ExposureNotification.android.cs

```diff
@@ -215,6 +240,125 @@ internal static async Task<ExposureDetectionSummary> PlatformGetExposureSummaryA
+				throw new InvalidOperationException("The handler does not support Exposure Window Mode.");
+
+			var config = await DailySummaryHandler.GetDailySummaryConfigurationAsync();
+			if (config == null)
+				throw new InvalidOperationException("The daily summary configuration was not provided on the handler.");
+
+			var builder = new DailySummariesConfig.DailySummariesConfigBuilder();
+
+			var attenuationThresholds = config.AttenuationThresholds.Select(t => (Java.Lang.Integer)t).ToArray();
+			var attenuationWeights = new Java.Lang.Double[4];
+			var aW = config.AttenuationWeights;
+			attenuationWeights[0] = new Java.Lang.Double(aW[DistanceEstimate.Immediate]);
+			attenuationWeights[1] = new Java.Lang.Double(aW[DistanceEstimate.Near]);
+			attenuationWeights[2] = new Java.Lang.Double(aW[DistanceEstimate.Medium]);
+			attenuationWeights[3] = new Java.Lang.Double(aW[DistanceEstimate.Other]);
+
+			builder.SetAttenuationBuckets(attenuationThresholds, attenuationWeights);
+
+			builder.SetDaysSinceExposureThreshold(config.DaysSinceLastExposureThreshold);
+
+			foreach (var pair in config.InfectiousnessWeights)
+			{
+				builder.SetInfectiousnessWeight(pair.Key.ToNative(), pair.Value);
+			}
+
+			foreach (var pair in config.ReportTypeWeights)
+			{
+				builder.SetReportTypeWeight(pair.Key.ToNative(), pair.Value);
+			}
+
+			var summaries = await GetClient().GetDailySummariesAsync(builder.Build());
+			if (summaries == null || summaries.Count == 0)
+				return Array.Empty<DailySummary>();
+
+			return summaries.Select(s => new DailySummary(
+				DateTime.UnixEpoch + TimeSpan.FromDays(s.DaysSinceEpoch),
+				s.SummaryData.FromNative(),
+				new Dictionary<ReportType, DailySummaryReport?>
+				{
+					[ReportType.Unknown] = s.GetReport(ReportType.Unknown),
+					[ReportType.ConfirmedTest] = s.GetReport(ReportType.ConfirmedTest),
+					[ReportType.ConfirmedClinicalDiagnosis] = s.GetReport(ReportType.ConfirmedClinicalDiagnosis),
+					[ReportType.SelfReported] = s.GetReport(ReportType.SelfReported),
+					[ReportType.Recursive] = s.GetReport(ReportType.Recursive),
+					[ReportType.Revoked] = s.GetReport(ReportType.Revoked),
+				}));
+		}
+
+		internal static async Task<IEnumerable<ExposureWindow>> PlatformGetExposureWindowsAsync()
+		{
+			if (DailySummaryHandler == null)
+				throw new InvalidOperationException("The handler does not support Exposure Window Mode.");
+
+			var windows = await GetClient().GetExposureWindowsAsync();
+			if (windows == null || windows.Count == 0)
+				return Array.Empty<ExposureWindow>();
+
+			return windows.Select(w => new ExposureWindow(
+				w.CalibrationConfidence.FromNativeCalibrationConfidence(),
+				DateTime.UnixEpoch + TimeSpan.FromMilliseconds(w.DateMillisSinceEpoch),
+				w.Infectiousness.FromNativeInfectiousness(),
+				w.ReportType.FromNativeReportType(),
+				w.ScanInstances.Select(s => s.FromNative())));
+		}
+
+		internal static async Task PlatformUpdateDiagnosisKeysDataMappingAsync()
```

Comment:
I couldn't find the code calling this method in this PR...

Reply:
Could you search PlatformGetExposureWindowsAsync ?

internal static async Task<IEnumerable<ExposureWindow>> PlatformGetExposureWindowsAsync()

Reply:
I'm sorry. I might have confused you.
Where is `PlatformUpdateDiagnosisKeysDataMappingAsync` called?

Reply:
You are correct, @tmurakami, this is not called in the nuget code. We ended up into calling `SetDiagnosisKeysDataMappingAsync` manually before calling `UpdateKeysFromServer`. Ideally, this should be called on Android after getting keys and configuration, so it won't send one more request to the server to fetch configuration one more time. There is, apparently, a limit on number of time how often this API can be called, so I would recommend either catching the exception when this limit is reached, or ensure that the method is not called more often than once a week ([src](https://developers.google.com/android/exposure-notifications/meaningful-exposures#map-diag-keys))

Reply:
@pavshr Thank you for sharing the information :heart: